### PR TITLE
fix(FEC-8714): removing static video element from the engine

### PR DESCRIPTION
**Pull request by odedhutzler**
### Description of the Changes

If there are two players playing autoplay, then only the first video element created is played.
Removed the use of static video element, left only the one within the engine instance.i

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/engines/html5/html5.js [hunk 1]
```diff
@@ -172,7 +171,7 @@ export default class Html5 extends FakeEventTarget implements IEngine {
     super();
     this._eventManager = new EventManager();
     this._canLoadMediaSourceAdapterPromise = Promise.resolve();
-    this._createVideoElement();
+    this._createVideoElement(config.tagetId);
```
[the remark this answers is not in the document]
Comment (OrenMe) [reply src/engines/html5/html5.js]: We have _playerId, but it is created only later in the constructor when calling _createPlayerContainer. Check if we can move it up.

File: src/player.js [hunk 2]
```diff
@@ -480,6 +480,12 @@ export default class Player extends FakeEventTarget {
     }
   }
 
+  _setConfigLogLevel(config: Object) {
```
Comment (dan-ziv) [inline src/player.js]: Move to private API section in the file

File: src/player.js [hunk 3]
```diff
@@ -181,12 +181,12 @@ export default class Player extends FakeEventTarget {
   /**
    * For browsers which block auto play, use the user gesture to open the video element and enable playing via API.
    * @returns {void}
+   * @param {string} playerId - the target id of the player
    * @private
-   * @static
    */
-  static _prepareVideoElement(): void {
+  _prepareVideoElement(): void {
```
Comment (dan-ziv) [inline src/player.js]: Move to private API section in the file.

File: src/player.js [hunk 4]
```diff
@@ -181,12 +181,12 @@ export default class Player extends FakeEventTarget {
   /**
    * For browsers which block auto play, use the user gesture to open the video element and enable playing via API.
    * @returns {void}
+   * @param {string} playerId - the target id of the player
```
Comment (dan-ziv) [inline src/player.js]: the id of the player (this is not the target id)

File: src/engines/html5/html5.js [hunk 5]
```diff
@@ -74,12 +74,9 @@ export default class Html5 extends FakeEventTarget implements IEngine {
   static id: string = 'html5';
 
   /**
-   * A video element for browsers which block auto play.
-   * @type {HTMLVideoElement}
-   * @private
-   * @static
+   * @type {PKVideoElementStore} - mapping between playerId to the video element.
```
Comment (dan-ziv) [inline src/engines/html5/html5.js]: Store object which mapping between playerId to its video element

File: src/player.js [hunk 6]
```diff
@@ -773,6 +779,7 @@ export default class Player extends FakeEventTarget {
   set muted(mute: boolean): void {
     if (this._engine) {
       this._engine.muted = mute;
+      Player.runCapabilities();
```
Comment (dan-ziv) [inline src/player.js]: ?

File: src/player.js [hunk 7]
```diff
@@ -1345,6 +1333,24 @@ export default class Player extends FakeEventTarget {
...
+   * @param {string} playerId - the id of the player
+   * @private
+   */
+  _prepareVideoElement(): void {
+    EngineProvider.getEngines().forEach((Engine: typeof IEngine) => {
+      Engine.prepareVideoElement(this._playerId);
+    });
+  }
+
+  _setConfigLogLevel(config: Object) {
```
Comment (dan-ziv) [inline src/player.js]: add jsdoc

**Pull request merged**